fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/DataHandler.java

```diff
@@ -0,0 +1,16 @@
+package com.github.dedis.popstellar.utility.handler;
+
+import com.github.dedis.popstellar.model.network.method.message.data.Data;
+import com.github.dedis.popstellar.utility.error.DataHandlingException;
+
+@FunctionalInterface
+public interface DataHandler<T> {
+
+  void accept(HandlerContext context, T data) throws DataHandlingException;
+
+  @SuppressWarnings("unchecked")
+  default void accept(HandlerContext context, Data data)
+      throws DataHandlingException, ClassCastException {
+    accept(context, (T) data);
+  }
```

Comment:
This should be removed. I would rather give the Registry the responsibility to cast a data.

Maybe add a `handle()` function in the Registry rather than a `getHandler()`.